universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java

```diff
@@ -41,15 +42,17 @@
 public class ProgramBlockEditView extends BaseHtmlView {
 
   private final AdminLayout layout;
+  private final Config config;
 
   public static final String ENUMERATOR_ID_FORM_FIELD = "enumeratorId";
   private static final String CREATE_BLOCK_FORM_ID = "block-create-form";
   private static final String CREATE_REPEATED_BLOCK_FORM_ID = "repeated-block-create-form";
   private static final String DELETE_BLOCK_FORM_ID = "block-delete-form";
 
   @Inject
-  public ProgramBlockEditView(AdminLayout layout) {
+  public ProgramBlockEditView(AdminLayout layout, Config config) {
     this.layout = checkNotNull(layout);
+    this.config = checkNotNull(config);
```

Comment:
Suggestion (blocking): prefer to set a variable for the specific config setting rather than the whole config object i.e. 
```java
private final boolean renderOptionalToggle; 
...
this.renderOptionalToggle = checkNotNull(config).hasPath("cf.optional_questions");
```

Reply:
done